File: tut/6thu/phone.py
#!/usr/bin/python2.7 -u

names = {}
addresses = {}
phones = {}

with open("people.txt", "r") as f:
  for line in f:
    (key, name, street, suburb) = line.strip().split(",")
    names[key] = name
    addresses[key] = "%s, %s" % (street, suburb)

with open("phones.txt", "r") as f:
  for line in f:
    (key, type, phone) = line.strip().split(",")
    if key not in phones:
      phones[key] = {}
    phones[key][type] = phone

for key in sorted(names):
  phones_str = "???"
  if key in phones:
    phones_array = []
    for type in phones[key]:
      phones_array.append("%s (%s)" % (phones[key][type], type))
    phones_str = ", ".join(phones_array)
  print "%s\n%s\n%s\n\n" % (names[key], addresses[key], phones_str)

File: tut/6thu/phone.pl
#!/usr/bin/perl -w

%names = ();
%addresses = ();
%phones = ();

open(PEOPLE, "people.txt") or die("file does not exist.");
while ($line = <PEOPLE>) {
    chomp $line;
    (my $key, my $name, my $street, my $suburb) = split(/,/, $line);
    $names{$key} = $name;
    $addresses{$key} = "$street, $suburb";
}
close(PEOPLE);

open(PHONES, "phones.txt") or die ("I hate people.");
while ($line = <PHONES>) {
    chomp $line;
    ($key, $type, $phone) = split(/,/, $line);

    $phones{$key}{$type} = $phone;
}
close(PHONES);

foreach $key (sort(keys(%names))) {
    print "$names{$key}\n";
    print "$addresses{$key}\n";

    if (defined($phones{$key})) {
        @phones = ();
        foreach $type (sort(keys($phones{$key}))) {
            push @phones, "$phones{$key}{$type} ($type)";
        }
        print join(", ", @phones), "\n";
    } else {
        print "???\n";
    }

    print "\n";
}
